ktor-server/ktor-server-core/jvmAndNix/test/io/ktor/server/application/HooksTest.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.application
+
+import io.ktor.client.request.*
+import io.ktor.server.response.*
+import io.ktor.server.testing.*
+import io.ktor.utils.io.concurrent.*
+import kotlin.test.*
+
+class HooksTest {
+
+    @Test
+    fun testCustomHookExecuted() {
+        class HookHandler {
+            var called: Boolean by shared(false)
+        }
+
+        val handler = HookHandler()
+
+        val myHook = object : Hook<HookHandler.() -> Unit> {
+            override fun install(application: Application, config: HookHandler.() -> Unit) {
+                handler.apply(config)
+            }
+        }
+
+        val testPlugin = createApplicationPlugin("TestPlugin") {
+            on(myHook) {
+                called = true
+            }
+        }
+
+        assertFalse(handler.called)
+
+        testApplication {
+            install(testPlugin)
+
+            routing {
+                handle {
+                    call.respondText("OK")
+                }
+            }
+
+            client.get("/")
+        }
+
+        assertTrue(handler.called)
```

Comment:
Isn't it true after installing, even without making a request?

Reply:
yep, I can also add a test with request than